### DISCUSSION TITLE
hardcoded connection limit of 32768

### DISCUSSION THREAD

**Discussion by ei8ht187** [coinfoundry/miningcore#1429] [Q&A] [answered]
We recently had a log message that more than 32768 miners are connected. It seems that this limit is hardcoded https://github.com/oliverw/miningcore/blob/b0f872f0f8f2398e633e7d2d700976cc5b650d03/src/Miningcore/Stratum/StratumConnection.cs#L51

Is there any reason, except limit the ressources, for this hardcoded connection limit? If not, is it possible to make this value configurable? Asking because our servers can easily handle those 32k connections.

**Comment by oliverw** [comment, marked as answer]
MC has no artificial connection limit. The line of code you referred to defines the maximum length of a JSON-RPC request received by a miner.

You most like likely hit an OS limit.

**Reply by ei8ht187** [reply]
Just asking because we had several error logs caused by this line https://github.com/oliverw/miningcore/blob/b0f872f0f8f2398e633e7d2d700976cc5b650d03/src/Miningcore/Stratum/StratumConnection.cs#L265

This references back to the constant variable in my first post.  

Edit: Ah wait... you mean, there was someone sending answers bigger than 32k of something (bit/byte/characters)? or one miner with 32k connections from the same wallet?

**Reply by oliverw** [reply]
> Just asking because we had several error logs caused by this line
> 
> https://github.com/oliverw/miningcore/blob/b0f872f0f8f2398e633e7d2d700976cc5b650d03/src/Miningcore/Stratum/StratumConnection.cs#L265
> 
> This references back to the constant variable in my first post.
> 
> Edit: Ah wait... you mean, there was someone sending answers bigger than 32k of something (bit/byte/characters)? or one miner with 32k connections from the same wallet?

The former (bytes).
